test/tests/workspace/gp_top_test.go

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2020 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License.AGPL.txt in the project root for license information.
+
+package workspace
+
+import (
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	agent "github.com/gitpod-io/gitpod/test/pkg/agent/workspace/api"
+	"github.com/gitpod-io/gitpod/test/pkg/integration"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/pkg/features"
+)
+
+func TestGpTop(t *testing.T) {
+	f := features.New("gp top").
+		WithLabel("component", "workspace").
+		WithLabel("type", "gp top").
+		Assess("it can run gp top and retrieve cpu/memory usage", func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+			t.Parallel()
+
+			ctx, cancel := context.WithTimeout(ctx, time.Duration(5*time.Minute))
+			defer cancel()
+
+			api := integration.NewComponentAPI(ctx, cfg.Namespace(), kubeconfig, cfg.Client())
+			t.Cleanup(func() {
+				api.Done(t)
+			})
+
+			nfo, stopWs, err := integration.LaunchWorkspaceDirectly(t, ctx, api)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			t.Cleanup(func() {
+				sctx, scancel := context.WithTimeout(context.Background(), 5*time.Minute)
+				defer scancel()
+
+				sapi := integration.NewComponentAPI(sctx, cfg.Namespace(), kubeconfig, cfg.Client())
+				defer sapi.Done(t)
+
+				if _, err = stopWs(true, sapi); err != nil {
+					t.Errorf("cannot stop workspace: %q", err)
+				}
+			})
+
+			rsa, closer, err := integration.Instrument(integration.ComponentWorkspace, "workspace", cfg.Namespace(), kubeconfig, cfg.Client(), integration.WithInstanceID(nfo.Req.Id))
+			integration.DeferCloser(t, closer)
+			if err != nil {
+				t.Fatalf("unexpected error instrumenting workspace: %v", err)
+			}
+			defer rsa.Close()
+
+			t.Logf("running gp top")
+			var res agent.ExecResponse
+			err = rsa.Call("WorkspaceAgent.Exec", &agent.ExecRequest{
+				Dir:     "/workspace",
+				Command: "gp",
+				Env:     []string{"SUPERVISOR_ADDR=10.0.5.2:22999"},
```

Comment:
Can we use another IP here? (not sure we can assume the same range in dedicated)

Reply:
That's the address supervisor is listening on in the preview env integration tests, this env var just points `gp` to talk to that address (other tests do this as well). Let's revisit this address when trying to run the tests for dedicated, I'll make a note in the relevant issue